common/middleware/set-services.js

```diff
@@ -0,0 +1,21 @@
+const services = require('../services')
+
+const setServices = (req, res, next) => {
+  const servicesCache = {}
```

Comment:
I'm not sure I fully understand the benefit of this cache...is there a case where this middleware would be called twice during a request?

What is the cache trying to prevent?

Reply:
Making sure that `req.services.foo()` returns the same 'instance' no matter how many times it's called

Reply:
I think this is resolved with the approach of instantiating the services here so the controllers always access whatever is set here, no service is 'called' in multiple places.

This isn't implemented here yet though! Currently the services are still just objects with nothing passed.